cpp/examples/external_sorting/random_table_generator.cu

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "parquet_io.hpp"
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/random/linear_congruential_engine.h>
+#include <thrust/random/uniform_int_distribution.h>
+#include <thrust/random/uniform_real_distribution.h>
+#include <thrust/transform.h>
+
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+/**
+ * @file random_table_generator.cuh
+ * @brief Random table and column generation utilities for external sorting example
+ *
+ * This header provides utilities to generate random tables and columns with various
+ * data types for testing and demonstration purposes. The implementation uses thrust
+ * algorithms for GPU-accelerated generation.
+ */
+
+namespace cudf {
+namespace examples {
+
+/**
+ * @brief Generate a random numeric column using thrust
+ *
+ * Creates a column filled with random values of the specified numeric type.
+ * Uses thrust::uniform_int_distribution for integral types and
+ * thrust::uniform_real_distribution for floating-point types.
+ *
+ * @tparam T The numeric data type (int32_t, int64_t, float, double, etc.)
+ * @param lower Lower bound of the random values (inclusive)
+ * @param upper Upper bound of the random values (inclusive for int, exclusive for float)
+ * @param num_rows Number of rows in the output column
+ * @param stream CUDA stream for device operations
+ * @param mr Device memory resource for allocations
+ * @return Unique pointer to the generated column
+ */
+template <typename T>
+std::unique_ptr<cudf::column> generate_random_numeric_column(T lower,
+                                                             T upper,
+                                                             cudf::size_type num_rows,
+                                                             rmm::cuda_stream_view stream,
+                                                             rmm::device_async_resource_ref mr)
+{
+  auto col = cudf::make_numeric_column(
+    cudf::data_type{cudf::type_to_id<T>()}, num_rows, cudf::mask_state::ALL_VALID, stream, mr);
```

Comment:
If we don't want the col to be nullable, we can also just do unallocated, right?